widgets/bat_openbsd.lua

```diff
@@ -19,67 +19,71 @@
 
 -- {{{ Grab environment
 local tonumber = tonumber
-local math = { floor = math.floor, modf = math.modf }
+local math = {
+    floor = math.floor,
+    modf = math.modf
+}
 
-local helpers = require"vicious.helpers"
-local spawn = require"vicious.spawn"
+local helpers = require("vicious.helpers")
 -- }}}
 
-local STATES = { [0] = "↯",         -- not charging
-                 [1] = "-",         -- discharging
-                 [2] = "!",         -- critical
-                 [3] = "+",         -- charging
-                 [4] = "N/A",       -- unknown status
-                 [255] = "N/A" }    -- unimplemented by the driver
+local bat_openbsd = {}
+function bat_openbsd.async(format, warg, callback)
+    local battery_id = warg or "bat0"
 
-return helpers.setasyncall{
-    async = function (format, warg, callback)
-        local filter = "hw.sensors.acpi" .. (warg or "bat0")
-        local pattern = filter .. ".(%S+)=(%S+)"
-        local bat_info = {}
+    local fields = {
+	charging_rate = ("hw.sensors.acpi%s.power0"):format(battery_id),
+	last_full_capacity = ("hw.sensors.acpi%s.watthour0"):format(battery_id),
+	remaining_capacity = ("hw.sensors.acpi%s.watthour3"):format(battery_id),
+	design_capacity = ("hw.sensors.acpi%s.watthour4"):format(battery_id),
+	state = ("hw.sensors.acpi%s.raw0"):format(battery_id)
+    }
 
-        spawn.with_line_callback_with_shell(
-            ("sysctl -a | grep '^%s'"):format(filter),
-            { stdout = function (line)
-                  for key, value in line:gmatch(pattern) do
-                      bat_info[key] = value
-                  end
-              end,
-              output_done = function ()
-                  -- current state
-                  local state = STATES[tonumber(bat_info.raw0)]
+    local sysctl_args = {}
+    for _, v in pairs(fields) do table.insert(sysctl_args, v) end
 
-                  -- battery capacity in percent
-                  local percent = tonumber(
-                      bat_info.watthour3 / bat_info.watthour0 * 100)
+    local battery = {}
+    helpers.sysctl_async(sysctl_args, function(ret)
```

Comment:
Please add the space before the parameter list of the anonymous function like what you did with #92.